gapic/templates/%namespace/%name_%version/%sub/services/%service/_mixins.py.j2

```diff
@@ -0,0 +1,409 @@
+
+        # Wrap the RPC method; this adds retry and timeout information,
+        # and friendly error handling.
+        rpc = gapic_v1.method.wrap_method(
+            self._transport.get_location,
+            default_timeout=None,
+            client_info=DEFAULT_CLIENT_INFO,
+        )
+
+        # Certain fields should be provided within the metadata header;
+        # add these here.
+        metadata = tuple(metadata) + (
+            gapic_v1.routing_header.to_grpc_metadata(
+                (("name", request.name),)),
+        )
+
+        # Send the request.
+        response = rpc(
+            request, retry=retry, timeout=timeout, metadata=metadata,)
+
+        # Done; return the response.
+        return response
+    {% endif %}
+
+    {% if "ListLocations" in api.mixin_api_methods %}
+    def list_locations(
+        self,
+        request: locations_pb2.ListLocationsRequest = None,
+        *,
+        retry: OptionalRetry = gapic_v1.method.DEFAULT,
+        timeout: float = None,
+        metadata: Sequence[Tuple[str, str]] = (),
+    ) -> locations_pb2.ListLocationsResponse:
+        r"""Lists information about the supported locations for this service.
+
+        Args:
+            request (:class:`~.location_pb2.ListLocationsRequest`):
+                The request object. Request message for
+                `ListLocations` method.
+            retry (google.api_core.retry.Retry): Designation of what errors,
+                 if any, should be retried.
+            timeout (float): The timeout for this request.
+            metadata (Sequence[Tuple[str, str]]): Strings which should be
+                sent along with the request as metadata.
+        Returns:
+            ~.location_pb2.ListLocationsResponse:
+                Response message for ``ListLocations`` method.
+        """
+        # Create or coerce a protobuf request object.
+        # The request isn't a proto-plus wrapped type,
+        # so it must be constructed via keyword expansion.
+        if isinstance(request, dict):
+            request = locations_pb2.ListLocationsRequest(**request)
+
+        # Wrap the RPC method; this adds retry and timeout information,
+        # and friendly error handling.
+        rpc = gapic_v1.method.wrap_method(
+            self._transport.list_locations,
+            default_timeout=None,
+            client_info=DEFAULT_CLIENT_INFO,
+        )
+
+        # Certain fields should be provided within the metadata header;
+        # add these here.
+        metadata = tuple(metadata) + (
+            gapic_v1.routing_header.to_grpc_metadata(
+                (("name", request.name),)),
+        )
+
+        # Send the request.
+        response = rpc(
+            request, retry=retry, timeout=timeout, metadata=metadata,)
+
+        # Done; return the response.
+        return response
+    {% endif %}
+    {% endif %}
+
+    {# LRO mixin #}
+
+    {% if api.has_operations_mixin() %}
+    def operations_client(self):
```

Comment:
There is no requirement to have this method. As we discussed, it is up to language to determine what might work best. For Go, we will put the List, Get, etc. Operations methods on the surface of the client lib, rather than expose the wrapped Operations client. C# went with the approach presented here, expose an accessor for the wrapped method. Your experience with Java might help Aza weigh the choices. I don't think it is a bad design, if the returned client is transport-agnostic.

Reply:
Sounds good. I'll remove LRO mixin (i.e. check for api.has_operations_mixin) for now. It can be added in the future, if deemed useful.

Reply:
@vam-google so actually after reading @noahdietz comment again, I realized we may have misinterpreted what he said. While not necessary to have `def operations_client`, it's a requirement to have an LRO mixin support. 

Not exposing `operations_client`, means we must go with approach Go used - put List, Get LRO methods in the surface. Practically that just basically means adding more copy-pasted code to `_mixins.py.j2`. 

I went with the current approach because I didn't want to repeat all of the code from [api_core](https://github.com/googleapis/python-api-core/blob/main/google/api_core/operations_v1/operations_client.py). However, I can see how exposing internals of the transport (i.e. transport._operations_client) violates encapsulation and may cause more harm in the future. Also I guess I've already copy-pasted a ton of stuff, so wouldn't be a huge deal to add LRO code.

Finally, my plan is to add individual LRO methods instead of `operations_client`.

@vam-google WDYT?

Reply:
Chatted offline - since LRO methods are an additive feature, let's code it in the next PR if needed.